File: app/src/main/java/com/example/nearby/adapter/MyOfferAdapter.java
package com.example.nearby.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.nearby.R;
import com.example.nearby.model.Coupon;

import java.util.ArrayList;
import java.util.List;

public class MyOfferAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<Coupon> items = new ArrayList<>();
    private OnItemClickListener mOnItemClickListener;
    private Context ctx;

    public interface OnItemClickListener{
        public void onItemClick(View view, Coupon obj, int position);
    }
    public void setmOnItemClickListener(OnItemClickListener mOnItemClickListener){
        this.mOnItemClickListener = mOnItemClickListener;
    }

    public MyOfferAdapter(Context context){
        ctx = context;
    }
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerView.ViewHolder vh;
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.indiv_my_offer,parent,false);
        vh = new OriginalViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
          Coupon obj = items.get(position);
          if(holder instanceof OriginalViewHolder){
              OriginalViewHolder view = (OriginalViewHolder) holder;
              view.offerPrice.setText(String.valueOf(obj.getPrice()));
              view.shopName.setText(obj.getShopName());
            /*  view.purchasedDate.setText();*/
          }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void setData(List<Coupon> couponList){
        items.clear();
        items.addAll(couponList);
        notifyDataSetChanged();
    }

    public class OriginalViewHolder extends RecyclerView.ViewHolder{
        TextView shopName;
        TextView purchasedDate;
        TextView offerPrice;
        ImageView imageView;
        public OriginalViewHolder(@NonNull View itemView) {
            super(itemView);
            shopName = itemView.findViewById(R.id.shopName);
            purchasedDate = itemView.findViewById(R.id.purchasedDate);
            offerPrice = itemView.findViewById(R.id.offerAmount);
            imageView = itemView.findViewById(R.id.purchasedOfferImage);

        }
    }


}

File: app/src/main/java/com/example/nearby/adapter/MyOfferAdapter.kt
package com.example.nearby.adapter

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.nearby.R
import com.example.nearby.model.IndivOfferPurchased
import com.example.nearby.utils.Tools

class MyOfferAdapter(val context : Context):RecyclerView.Adapter<MyOfferAdapter.ViewHolder>() {

    var mList = ArrayList<IndivOfferPurchased>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyOfferAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.indiv_my_offer,parent,false)
        return ViewHolder(view)
    }


    override fun onBindViewHolder(holder: MyOfferAdapter.ViewHolder, position: Int) {
        val offerPurchased = mList[position]


        holder.price.text = offerPurchased.coupon.price.toString()
        holder.purchasedDate.text = offerPurchased.transactionDate
        holder.couponCode.text = offerPurchased.coupon.code
        holder.shopName.text = offerPurchased.coupon.shopName
        Tools.displayImageOriginal(context,holder.image,offerPurchased.coupon.image)

    }

    class ViewHolder(ItemView : View):RecyclerView.ViewHolder(ItemView) {
        val image : ImageView = itemView.findViewById(R.id.purchasedOfferImage)
        val shopName : TextView = itemView.findViewById(R.id.shopName)
        val price : TextView = itemView.findViewById(R.id.offerAmount)
        val purchasedDate : TextView = itemView.findViewById(R.id.purchasedDate)
        val couponCode : TextView = itemView.findViewById(R.id.code)
    }

    override fun getItemCount(): Int {
       return mList.size
    }

    fun setData(offersAvailedList : List<IndivOfferPurchased>):Unit{
        mList.clear()
        mList.addAll(offersAvailedList)
        notifyDataSetChanged()
    }
}